webclient/ez/views.py

```diff
@@ -382,6 +384,33 @@ def init_translation(self):
         self.button = _("ez.templates.ExterniZdrojPotvrditView.submitButton.text")
         self.success_message = EZ_USPESNE_POTVRZEN
 
+    def post(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        zaznam: ExterniZdroj = context["object"]
+        fedora_transaction = zaznam.create_transaction(request.user, self.success_message)
+        try:
+            zaznam.save()
+            zaznam.close_active_transaction_when_finished = True
+            getattr(ExterniZdroj, self.action)(zaznam, request.user)
+            return JsonResponse({"redirect": zaznam.get_absolute_url()})
+        except (DoiWriteError, FedoraError) as err:
+            logger.info(
+                "ez.models.ExterniZdroj.set_potvrzeny.error", extra={"error": err, "ident_cely": zaznam.ident_cely}
+            )
+            from arch_z.models import ArcheologickyZaznam
+
+            for akce in zaznam.externi_odkazy_zdroje.all():
+                if (
+                    akce.archeologicky_zaznam.typ_zaznamu == ArcheologickyZaznam.TYP_ZAZNAMU_LOKALITA
+                    and akce.archeologicky_zaznam.stav == AZ_STAV_ARCHIVOVANY
+                    and akce.archeologicky_zaznam.lokalita.igsn
+                ):
+                    akce.archeologicky_zaznam.igsn_lokalita_update(False)
+            transaction.set_rollback(True)
+            fedora_transaction.rollback_transaction()
+            transaction.set_rollback(True)
```

Comment:
Cyklus by se myslím měl posunout až za tento rollback.

Pokud se nepletu, tak dokud se nepustí ten rollback, tak bude záznam EZ ve stavu před ním a tím pádem se všechny závislosti budou vyhodnocovat jakoby než dojde k rollbacku (jsou tam různé podmínky, které závisí např. na stavu záznamu apod.). Je ale možné, že prostě to chování chybně interpretuju, tak se případně omlouvám, že s tím furt prudím.

Reply:
Ok, ale default je `reload_record=False` a tady se to nijak nemění na `true`...